### ISSUE TITLE
[TraceQL] Support quoted attribute names

### ISSUE THREAD

**Issue by joe-elliott** [grafana/grafana#77394]
Tempo now has the ability to support quoted attributes. This allows users to query attribute names that have spaces and other esoteric characters.

This is now a valid way to query the `span.db.statement` field:

```
{ span."db.statement" != "" }
```

There are two escape sequences `\"` and `\\` for obvious reasons.

Currently Grafana just highlights this as a string and does not correctly show autocomplete:

![image](https://github.com/grafana/grafana/assets/2272392/c0c53731-145d-4c72-bc8e-2e9327a7d5dc)

Quotes can be started and ended at any time inside of the attribute string so this is technically valid:
```
{ span.db.st"ateme"nt != "" }
```

**Comment by ifrost** [comment]
cc @grafana/observability-traces-and-profiling

**Comment by joe-elliott** [comment]
Yes, I agree that we should correctly return values on `api/v2/search/tag/span.ht"tp.met"hod/values`.  @mapno can you make an issue for this and track it? I didn't realize that Grafana was correctly sending the name as typed.

For Grafana's side maybe this is just a syntax highlighting issue?
